### PR TITLE
Update 2020.07 kt

### PR DESCRIPTION
**Pull request by iamzoltan**
I think I have finished the notebook. There seems to be an error when github loads the markdown for one section. Do any of you know how to fix it? Please give me feedback :)

### CODE REVIEW

File: week1_session4_intro_to_ml_and_scikit_learn.ipynb [hunk 1]
```diff
@@ -0,0 +1,1547 @@
+{
```
Comment (dominiquesydow) [inline week1_session4_intro_to_ml_and_scikit_learn.ipynb:1]: <p>In order to follow the scheme we talked about during the last meeting, i.e.</p><p> </p><p>1. Aims of this notebook</p><p>2. Learning goals (Theory and Practical)</p><p>3. References</p><p>4. Theory</p><p>5. Practical</p><p>6. Discussion</p><p>7. Exercises.</p><p>8. Solutions</p><p> </p><p>you could split this cell, to fill the categories:</p><p> </p><p>1. Aims of this notebook</p><blockquote>We want to introduce you to the basic concepts of machine learning and how to use python and scikit-learn to apply these concepts.</blockquote><p>Maybe mention here already the dataset you will be using.</p><p> </p><p>2. Learning goals</p><blockquote>By the end of this session, you should be familiar with:</blockquote><blockquote>the difference between supervised and unsupervised learning:</blockquote><blockquote class="ql-indent-1">classification vs regression in supervised learning</blockquote><blockquote>loading, exploring and preprocessing a dataset; specifically a csv file</blockquote><blockquote>how and why we split data into training and test sets</blockquote><blockquote>training a model on the training set</blockquote><blockquote>evaluating the performance of a model on the test set</blockquote><blockquote>using various performance measures to validate a model</blockquote><p> </p><p>3. References</p><p>TBA</p><p> </p><p>4. Theory</p><p>The theory will be covered separately with slides (link).</p><p>Maybe add a TOC</p><blockquote><strong>note:</strong> see slides for an overview of the first topic.</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/volkamerlab/ai_in_medicine/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (dominiquesydow) [inline week1_session4_intro_to_ml_and_scikit_learn.ipynb:1]: <p>Above this cell add </p><p> </p><h2>5. Practical</h2>
<br/>

 _Reply via <a href='https://app.reviewnb.com/volkamerlab/ai_in_medicine/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (dominiquesydow) [inline week1_session4_intro_to_ml_and_scikit_learn.ipynb:1]: <p>Now that I see this, I might add this to the pandas notebook also (<code>iloc</code>/<code>loc</code>),  so that they have heard it before.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/volkamerlab/ai_in_medicine/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (dominiquesydow) [inline week1_session4_intro_to_ml_and_scikit_learn.ipynb:1]: <p>Maybe show one example, where they can see an assertion error? Just for fun :)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/volkamerlab/ai_in_medicine/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (dominiquesydow) [inline week1_session4_intro_to_ml_and_scikit_learn.ipynb:1]: <p>I suggest to define accuracy (here or in the theory part (4. Theory)) - you probably do all this in the slides, so you could in the notebook maybe only add some kind of glossar for very important terms in ML?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/volkamerlab/ai_in_medicine/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

Comment (dominiquesydow) [inline week1_session4_intro_to_ml_and_scikit_learn.ipynb:1]: <p>Maybe add a header to this section similar as you did for "Accuracy"?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/volkamerlab/ai_in_medicine/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

Comment (dominiquesydow) [inline week1_session4_intro_to_ml_and_scikit_learn.ipynb:1]: <p>Are you focusing on specific metrics for the course? They could be a bit overwhelmed by all these terms.</p><p> </p><p>I would definitely leave them in for those how are interested to learn more but I would probably focus only on a subset.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/volkamerlab/ai_in_medicine/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Comment (dominiquesydow) [inline week1_session4_intro_to_ml_and_scikit_learn.ipynb:1]: <p>Maybe save results to <code>results/week1_session4_scikit-learn/</code>  instead of <code>images</code>  to make it easier to link results to the respective notebook? This we could streamline with other notebooks' results</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/volkamerlab/ai_in_medicine/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='42'/>

Comment (dominiquesydow) [inline week1_session4_intro_to_ml_and_scikit_learn.ipynb:1]: <p>I imagine that the work load here is quite high for one lesson. I would sort the exercises from "mandatory" to "optional", so that at the end of the lesson you can wrap up the core (and leave the rest for them to explore if interested).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/volkamerlab/ai_in_medicine/pull/10/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='51'/>

**Comment by dominiquesydow** [comment]
@iamzoltan, I like the scope of this notebook! 
They are guided nicely through the steps of ML (with two example models) and in the end they have the possibility to set it up themselves.

See my suggestions above!

File: week1_session4_intro_to_ml_and_scikit_learn.ipynb (hunk 1 above)
Comment (iamzoltan) [reply week1_session4_intro_to_ml_and_scikit_learn.ipynb:1]: <p>have made these changes :)</p>

Comment (iamzoltan) [reply week1_session4_intro_to_ml_and_scikit_learn.ipynb:1]: <p>i will most likely leave the solutions out since the steps are laid out for them above</p>

Comment (iamzoltan) [reply week1_session4_intro_to_ml_and_scikit_learn.ipynb:1]: <p>good idea :)</p>

Comment (iamzoltan) [reply week1_session4_intro_to_ml_and_scikit_learn.ipynb:1]: <p>also a good idea :) done</p>

Comment (iamzoltan) [reply week1_session4_intro_to_ml_and_scikit_learn.ipynb:1]: <p>will do :)</p>

Comment (iamzoltan) [reply week1_session4_intro_to_ml_and_scikit_learn.ipynb:1]: <p>i will focus on accuracy, and confusion matrices, then I will probably go over the rest briefly. We will see how the time goes haha.</p>

Comment (iamzoltan) [reply week1_session4_intro_to_ml_and_scikit_learn.ipynb:1]: <p>dones :)</p>

Comment (iamzoltan) [reply week1_session4_intro_to_ml_and_scikit_learn.ipynb:1]: <p>I could leave everything after accuracy as optional, but I dont think its too bad. What do you think?</p>

**Comment by iamzoltan** [comment]
<p>ill add your changes and add moritzs suggestions too. then ill commit and push an updated version :)</p>

File: week1_session4_intro_to_ml_and_scikit_learn.ipynb (hunk 1 above)
Comment (dominiquesydow) [reply week1_session4_intro_to_ml_and_scikit_learn.ipynb:1]: That sounds good! I also would not remove exercises, since some might be quick and interested.

**Comment by iamzoltan** [comment]
I have implemented all the changes and added the missing sections. Let me know if there is any other issues please!
